### ISSUE TITLE
xonsh + kitty and disappearing text

### ISSUE THREAD

**Issue by tkossak** [xonsh/xonsh#4678]
Maximizing and restoring kitty terminal makes xonsh output from previous command invisible

## xonfig

<details>

```
$ xonfig
+------------------+-----------------------+
| xonsh            | 0.11.0                |
| Git SHA          | 337cf25a              |
| Commit Date      | Nov 17 15:37:41 2021  |
| Python           | 3.9.9                 |
| PLY              | 3.11                  |
| have readline    | True                  |
| prompt toolkit   | 3.0.28                |
| shell type       | prompt_toolkit        |
| history backend  | json                  |
| pygments         | 2.11.2                |
| on posix         | True                  |
| on linux         | True                  |
| distro           | manjaro               |
| on wsl           | False                 |
| on darwin        | False                 |
| on windows       | False                 |
| on cygwin        | False                 |
| on msys2         | False                 |
| is superuser     | False                 |
| default encoding | utf-8                 |
| xonsh encoding   | utf-8                 |
| encoding errors  | surrogateescape       |
| on jupyter       | False                 |
| jupyter kernel   | None                  |
| xontrib          | []                    |
| RC file 1        | /home/kossak/.xonshrc |
+------------------+-----------------------+

```

</details>

## Expected Behavior
Xonsh output from previous commands is still visible, as it was before maximizing and restoring kitty window.

## Current Behavior

Xonsh output disappears from terminal after maximinzing and restoring it. Example:

Before maximize/restore:
![image](https://user-images.githubusercontent.com/3728245/155005150-49f984c2-c4a8-4212-be0c-80f1f9ee0776.png)

After maximizing and restoring the window:
![image](https://user-images.githubusercontent.com/3728245/155005180-c1b28e89-b3b9-483d-9d85-11955bf64972.png)

I noticed that it doesn't actually disappear forever, but xonsh somehow inserts a lot of empty lines so I need to scroll about a full screen to see the output, that was visibile before maximize/restore.

I use:

shell: xonsh 0.11.0
terminal: kitty 0.24.1
Linux Manjaro KDE

## Steps to Reproduce
1. Open kitty terminal in small window (not maximized)
2. Enter and execute few commands eg:
```
echo 123
echo 456
```
3. Maximize window and then restore it to previous size
4. Observe that the output from previous commands disappeared

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by Techcable** [comment]
I'm curious because I use xonsh + kitty too.

Can you either post a gist with your `.xonshrc` or give the minimal `.xonshrc` that can reproduce this issue?

Also are you:
1. Outputing binary info into the console?
2. Using the `term-integration` xontrib ([here](https://github.com/jnoortheen/xontrib-term-integrations))?

**Comment by gforsyth** [comment]
Hey @kossak -- can you see if this issue occurs with the readline backend? I think this is probably related to prompt toolkit redrawing the window but we should confirm.

On Sat, Feb 26, 2022, at 11:51 AM, Kossak wrote:
> 
> 
> 
> I just checked and it happens also without any `.xonshrc` file (renamed my `~/.xonshrc` to `~/.xonshrc.old` and the problem still persists).
> I also renamed my `/home/kossak/.config/kitty/kitty.conf`, started kitty/xonsh again and problem still happens, so this config seems to have no effect on it either.
> 
> ad 1) I did not output binary info into the console, only echo the string. But I just tried to and the effect is the same (without any `.xonshrc` or `kitty.conf` files):
> 
> 
> before maximizing:
> image <https://user-images.githubusercontent.com/3728245/155851391-154dafba-e53a-41f0-b55d-2052cbe6b20f.png> image <https://user-images.githubusercontent.com/3728245/155851393-a64e9df5-7368-47ee-8a97-e2ef408ceca4.png>
> 
> after maximing and restoring window: image <https://user-images.githubusercontent.com/3728245/155851400-463129ab-97d5-40ee-9691-a55c80449ed3.png>
> 
> ad 2) No, I do not use `term-integration`, my xontrribs, as listed by `xontrib list`, are:
> 
> `abbrevs             installed      not-loaded
> autovox             installed      not-loaded
> bashisms            installed      not-loaded
> coreutils           installed      not-loaded
> distributed         installed      not-loaded
> free_cwd            installed      not-loaded
> jedi                installed      not-loaded
> kitty               installed      not-loaded
> mpl                 installed      not-loaded
> mplhooks            installed      not-loaded
> pdb                 installed      not-loaded
> prompt_ret_code     installed      not-loaded
> vox                 installed      not-loaded
> voxapi              installed      not-loaded
> whole_word_jumping  installed      not-loaded
> xog                 installed      not-loaded
> z                   installed      not-loaded
`

**Comment by kossak** [comment]
this was meant to @tkossak I believe :)

**Comment by gforsyth** [comment]
oops!  sorry @kossak !  I was on email and clearly can't be trusted without autocomplete.

**Comment by tkossak** [comment]
> Hey @kossak -- can you see if this issue occurs with the readline backend? I think this is probably related to prompt toolkit redrawing the window but we should confirm.

You're right - the readline xonsh doesn't have this problem! Is it prompt_toolkit's fault? Can I fix it myself with some xonshrc setting (besides switching to readline)?

**Comment by gforsyth** [comment]
Hey @tkossak -- it's definitely related to `prompt_toolkit` but I'm not sure if it's something we're doing in our usage of `ptk` or if it's purely an upstream issue.  

@jonathanslenders may be able to provide some more context

**Comment by tkossak** [comment]
Not sure it this is helpfull or not, but I tried several different versions of prompt_toolkit and versions >= 3.0.4 (with the same version of xonsh - 0.11.0) all have this problem. Prompt_toolkit == 3.0.3 does not work with this version of xonsh.

**Comment by tkossak** [comment]
I could dump prompt_toolkit for good in favor of readline, but I can't find a way to speicify readline hotkeys. Is it possible to define custom hotkeys for readline version of xonsh?
